pkg/kubernetes/pullsecrets.go

```diff
@@ -0,0 +1,44 @@
+package kubernetes
+
+import (
+	"context"
+
+	"github.com/loft-sh/devpod-provider-kubernetes/pkg/docker"
+	perrors "github.com/pkg/errors"
+)
+
+func (k *kubernetesDriver) EnsurePullSecret(
+	ctx context.Context,
+	pullSecretName string,
+	dockerCredentials *docker.Credentials,
+) error {
+	//todo: check if pull secret already exists, don't create it again
+	err := k.createPullSecret(ctx, pullSecretName, dockerCredentials)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (k *kubernetesDriver) createPullSecret(
+	ctx context.Context,
+	pullSecretName string,
+	dockerCredentials *docker.Credentials,
+) error {
+	args := []string{
+		"create",
+		"secret",
+		"docker-registry",
+		pullSecretName,
+		"--docker-username=" + dockerCredentials.Username,
+		"--docker-password=" + dockerCredentials.Secret,
+		"--docker-server=" + dockerCredentials.ServerURL,
+	}
```

Comment:
I would actually prefer the DevSpace way here, as username etc might be empty often, see above in the docker config.go. If you chose to go this way, we definitely need an e2e test for the case no username and username and password for this before we can merge your PR